[terra-tabs] Display tabs inline on smaller viewports.

nramamurth opened the pull request:
### Summary
* Added `responsiveTo` prop to provide ability to consumers to decide whether or not tabs need to completely collapse into a menu on smaller viewports or when tabs are placed within a very small container.
* Center-align text label in the single dropdown tab when all tabs completely collapse into a menu.

**Note**: "More" button looks off in the screenshot [here](https://github.com/cerner/terra-framework/pull/968/files?short_path=50d1105#diff-50d11059da76d878d7ed9c04aff27810) and I have logged a separate issue: https://github.com/cerner/terra-framework/issues/970. 

Closes #485, Closes #484

emilyrohrbough commented on packages/terra-tabs/src/Tabs.jsx:
```diff
@@ -169,35 +178,44 @@ class Tabs extends React.Component {
+          <ResponsiveElement
+            tiny={collapsedTabs}
+            small={collapsibleTabs}
+            responsiveTo={responsiveTo}
```
Will these changes be valid after https://github.com/cerner/terra-framework/pull/972 is merged?

nramamurth replied:
I don't think so; the core logic should probably remain as-is. I think what gets rendered now should be based off of both `breakpoint` value and `responsiveTo` value - something like —

```
if ((current breakpoint is "tiny") AND (responsiveTo is either "parent" or "window")) {
    <---- show collapsed tabs ---->
 } else {
    <---- show collapsible tabs ---->
}
```

emilyrohrbough commented on packages/terra-tabs/src/Tabs.jsx:
```diff
@@ -169,35 +178,44 @@ class Tabs extends React.Component {
-        {clonedPanes}
-      </CollapsibleTabs>
-    );
+    const header = (() => {
```
Can we rename this func to something like getTabs() to be representative of what it is returning?

emilyrohrbough replied:
or could we convert this to be something like:
```js
...
let tabContent = collapsibleTabs;

 if ([DependentViewport.PARENT, DependentViewport.WINDOW].includes(responsiveTo)) {
    ...
     tabContent (
          <ResponsiveElement
            tiny={collapsedTabs}
            small={collapsibleTabs}
            responsiveTo={responsiveTo}
      />
    )
 }

....

return (
      <ContentContainer
        {...customProps}
        className={tabsClassNames}
        fill={fill}
        header={tabContent}
      >
...
```

nramamurth replied:
I have renamed `header` ➡️ `tabContent` - https://github.com/cerner/terra-framework/pull/968/commits/e293226fe48790135e5279fb37bfa4a6a0f4fb35.

nramamurth commented:
@emilyrohrbough 

> We should had a window-responsive test example and generate wdio screenshots

Now that you mention it, I realized that my default parent-responsive spec wasn't really showing the difference between parent resize v/s window resize, so I have updated it and have also added a window-responsive spec: https://github.com/cerner/terra-framework/pull/968/commits/e293226fe48790135e5279fb37bfa4a6a0f4fb35

> Should we add add an example of these variants to the face-up doc examples?

That will surely help the consumers, me thinks. I shall add it.

mjhenkes commented on packages/terra-tabs/src/Tabs.jsx:
```diff
@@ -58,11 +58,19 @@ const propTypes = {
 const defaultProps = {
   tabFill: false,
   fill: false,
+  responsiveTo: DependentViewport.PARENT,
```
```suggestion
  responsiveTo: 'parent',
```
This will look better in the generated props table.

mjhenkes commented on packages/terra-tabs/src/Tabs.jsx:
```diff
@@ -169,35 +178,44 @@ class Tabs extends React.Component {
+        );
+
+        return (
+          <ResponsiveElement
```
If tabs are responsive to none, we should not wrap the tabs in a responsive element and just return `CollapsibleTabs`

mjhenkes replied:
Ah, i see you are doing this already, Disregard.

StephenEsser commented on packages/terra-tabs/src/Tabs.jsx:
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import classNames from 'classnames/bind';
 import ContentContainer from 'terra-content-container';
-import ResponsiveElement from 'terra-responsive-element';
+import ResponsiveElement, { DependentViewport } from 'terra-responsive-element';
```
I think the `DependentViewport` constant makes the code more challenging to understand and doesn't follow a consistent export strategy. There are only two options `window` and `parent`. I'd recommend just using the string literals. `window`, `parent`. 

I don't personally think keying objects in this context lowers the chances of typos/errors or adds enough reusability to adjust for the added complexity.

nramamurth replied:
> I don't personally think keying objects in this context lowers the chances of typos/errors or adds enough reusability to adjust for the added complexity.

It's been a grey area for me as to when to use it and when not to but this makes sense. I have [removed](https://github.com/cerner/terra-framework/pull/968/commits/4264fffd83d8119e411ddf2b7d1f7d303bf63c76) it.

nramamurth commented:
@emilyrohrbough 

Examples have been added to the site - https://github.com/cerner/terra-framework/pull/968/commits/d4333c8e2986f4fd631c694c0709461a4bbe5928
